apps/fyle/views.py

```
@@ -430,3 +436,110 @@ def post(self, request, *args, **kwargs):
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+class ExpenseView(generics.ListAPIView):
+    """
+    Expense view
+    """
+
+    serializer_class = ExpenseSerializer
+
+    def get_queryset(self):
+        start_date = self.request.query_params.get('start_date', None)
+        end_date = self.request.query_params.get('end_date', None)
+        org_id = Workspace.objects.get(id=self.kwargs['workspace_id']).fyle_org_id
+
+        filters = {
+            'org_id': org_id,
+            'is_skipped': True
+        }
+
+        if start_date and end_date:
+            filters['updated_at__range'] = [start_date, end_date]
+
+        queryset = Expense.objects.filter(**filters).order_by('-updated_at')
+
+        return queryset
+
+
+class CustomFieldView(generics.RetrieveAPIView):
+    """
+    Custom Field view
+    """
+    def get(self, request, *args, **kwargs):
+        """
+        Get Custom Fields
+        """
+        try:
+            workspace_id = self.kwargs['workspace_id']
+
+            fyle_credentails = FyleCredential.objects.get(workspace_id=workspace_id)
+
+            platform = PlatformConnector(fyle_credentails)
+
+            custom_fields = platform.expense_custom_fields.list_all()
+
+            response = [] 
+            response.extend(DEFAULT_FYLE_CONDITIONS)
+            for custom_field in custom_fields:
+                if custom_field['type'] in ('SELECT', 'NUMBER', 'TEXT'):
+                    response.append({
+                        'field_name': custom_field['field_name'],
+                        'type': custom_field['type'],
+                        'is_custom': custom_field['is_custom']
+                    })
+            
+            return Response(
+                data=response,
+                status=status.HTTP_200_OK
+            )
+
+        except InvalidTokenError:
+            logger.info('Invalid Fyle refresh token for workspace %s', workspace_id)
+
+        except Exception as exception:
```

broad exceptions are extremely incorrect, we should rather let it error out, there is not need for unnecessary try catches

---

yes shwetabh